sink/datadog.go

```diff
@@ -393,6 +394,11 @@ func (s *Datadog) sendApi(ddCtx context.Context, dp []datadogV2.MetricSeries) er
 			return err
 		}
 
+		if len(apiResponse.Errors) > 0 {
+			// datadog can return a 202 Accepted response code but errors in response payload
+			return fmt.Errorf("error response from Datadog: %s", strings.Join(apiResponse.Errors, ","))
```

Comment:
Re @Areson comment:  if this is _partial_ success, then we should not abort completely.